api_ekyc.go

```diff
@@ -50,69 +50,125 @@ type ResponseEkycLiveness struct {
 	Err  *Error            `json:"error"`
 }
 
-func (c Client) EkycMyKad(request RequestEkycMykad) (*ResponseEkycMykad, error) {
+type ResponseGetEkycResult struct {
+	Code string        `json:"code"`
+	Item GetEkycResult `json:"item"`
+	Err  *Error        `json:"error"`
+}
+
+// EkycMyKad :
+func (c Client) EkycMyKad(request RequestEkycMykad) (*ResponseEkycMykad, *Error) {
 	if c.err != nil {
-		return nil, c.err
+		return nil, &Error{
+			Code:    "INTERNAL_SERVER_ERROR",
+			Message: c.err.Error(),
+		}
 	}
 
 	response := new(ResponseEkycMykad)
 	requestURL := c.prepareAPIURL(pathAPIService)
-	if err := c.httpAPI(methodPOST, fmt.Sprintf("%s", requestURL), RequestService{
+	if err := c.httpAPI(methodPOST, requestURL, RequestService{
 		Service:  "ekyc",
 		Version:  "v1",
 		Function: "mykad",
 		Request:  request,
 	}, response); err != nil {
-		return nil, err
+		return nil, &Error{
+			Code:    "INTERNAL_SERVER_ERROR",
+			Message: err.Error(),
+		}
 	}
 
 	if response.Err != nil {
-		return response, errors.New(response.Err.Message)
+		return response, response.Err
 	}
 
 	return response, nil
 }
 
-func (c Client) EkycFaceCompare(request RequestEkycFaceCompare) (*ResponseEkycFaceCompare, error) {
+// EkycFaceCompare :
+func (c Client) EkycFaceCompare(request RequestEkycFaceCompare) (*ResponseEkycFaceCompare, *Error) {
```

Comment:
why the second return type is `*Error` instead of `error`? It's not recommended to use custom error as error